File: js/demo/objects/main-menu.js
'use strict';
import { Drawable, GameObject, override } from '../../engine';
import Home from '../rooms/home';
import Menu from './menu';

class MainMenu extends Menu {
  @override
  init() {
    this.opts = {
      ['New game']: () => {
        super.game.room.goto(Home);
      },
      ['Continue']: () => {
        super.game.restart();
      }
    };
  }
}

export default MainMenu;

File: c++/src/demo/objects/main-menu.cc
#include "main-menu.h"
#include "../rooms/home.h"

namespace Demo {
    MainMenu::MainMenu() : Menu {std::vector<Menu::Option>{
        {"New Game", [this] () { game().room_goto<RmHome>(); } },
        {"Continue", [this] () { game().restart(); } },
        {"Quit", [this] () { game().end(); } }
    }} {}
}
